pkg/chunk/aws_storage_client.go

```diff
@@ -331,28 +340,232 @@ func (a dynamoDBRequestAdapter) HasNextPage() bool {
 	return a.request.HasNextPage()
 }
 
-func (a awsStorageClient) GetChunk(ctx context.Context, key string) ([]byte, error) {
+func (a awsStorageClient) GetChunks(ctx context.Context, chunks []Chunk) ([]Chunk, error) {
+	var (
+		s3Chunks       []Chunk
+		dynamoDBChunks []Chunk
+	)
+
+	for _, chunk := range chunks {
+		if chunk.From.Before(a.cfg.ChunkTableFrom.Time) {
+			s3Chunks = append(s3Chunks, chunk)
+		} else {
+			dynamoDBChunks = append(dynamoDBChunks, chunk)
+		}
+	}
+
+	var err error
+	s3Chunks, err = a.getS3Chunks(ctx, s3Chunks)
+	if err != nil {
+		return nil, err
+	}
+
+	dynamoDBChunks, err = a.getDynamoDBChunks(ctx, dynamoDBChunks)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I don't expect us to be doing both simultaneously except for when we migrate, when it will only occur for a couple or hours.  So I didn't think it was worth the extra code to parallelise.  Will add comment to this effect.